lib/periodicChecks.js

```diff
@@ -0,0 +1,49 @@
+const assignUnassignedReviewers = async (pullRequest, context) => {
+  const assignees = pullRequest.assignees.map(user => user.login);
+  const reviewers = pullRequest.requested_reviewers.map(user => user.login);
+
+  // If the PR author is assigned, then don't assign other reviewers.
+  if (assignees.includes(pullRequest.user.login)) {
+    return;
+  }
+
+  let botComment = 'Hey ';
+  const prParams = context.repo(
+    {
+      number: pullRequest.number,
+    }
+  );
+
+  const unassignedReviewers = reviewers.filter(reviewer => {
+    if (!assignees.includes(reviewer)) {
+      botComment += `@${reviewer}, `;
+      return true;
+    }
+  });
+  if (unassignedReviewers.length > 0) {
+    await context.github.issues.addAssigneesToIssue(
+      { ...prParams, assignees: unassignedReviewers });
```

Comment:
What if the unassigned reviewer has already approved the PR? will it avoid adding those reviewers?

Reply:
If the reviewer has already approved the PR, they wouldn't get listed in `pullRequest.requested_reviewers`

Reply:
Can you please add this info as a comment around the line where you are using `requested_reviewers`?

Reply:
Done